### PR TITLE
Complete missing docstrings in keras-core/ops/core.py

### PR DESCRIPTION
**Pull request by Frightera**
Most of the py files have the functions defined in that module at top:

Example:
```python
"""
scatter
scatter_update
slice
slice_update
while_loop
stop_gradient
shape
cast
convert_to_tensor
convert_to_numpy
"""
```
Should they be removed?

### CODE REVIEW

File: keras_core/ops/core.py [hunk 1]
```diff
@@ -273,30 +273,128 @@ def stop_gradient(variable):
 
 @keras_core_export("keras_core.ops.shape")
 def shape(x):
-    """Gets the shape of the tensor input."""
+    """
+    Gets the shape of the tensor input.
+
+    Args:
+        x: A tensor. This function will try to access the `shape` attribute of
+            the input tensor. When using TensorFlow backend the input can also
```
Comment (fchollet) [inline keras_core/ops/core.py:281]: Avoid any backend-specific information. The behavior should be assumed identical in all backends.

File: keras_core/ops/core.py [hunk 2]
```diff
@@ -273,30 +273,128 @@ def stop_gradient(variable):
...
+            be a list.
+
+    Returns:
+        A tuple of integers, the shape of the tensor.
```
Comment (fchollet) [inline keras_core/ops/core.py:285]: Could include None entries as well.

File: keras_core/ops/core.py [hunk 3]
```diff
@@ -273,30 +273,128 @@ def stop_gradient(variable):
...
+
+    Examples:
+        TensorFlow Backend with a list:
```
Comment (fchollet) [inline keras_core/ops/core.py:288]: No code example should feature backend specific code.

Do something like:

```
>>> x = keras.ops.ones(shape=(3, 4))
>>> keras.ops.shape(x)
(3, 4)
```

File: keras_core/ops/core.py [hunk 4]
```diff
@@ -273,30 +273,128 @@ def stop_gradient(variable):
 
 @keras_core_export("keras_core.ops.shape")
 def shape(x):
-    """Gets the shape of the tensor input."""
+    """
+    Gets the shape of the tensor input.
```
Comment (fchollet) [inline keras_core/ops/core.py:277]: The one-liner should be the first line (the line above this)

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by Frightera** [comment]
@fchollet Should be fine now, I got the style now 😎

File: keras_core/ops/core.py [hunk 5]
```diff
@@ -273,30 +273,83 @@ def stop_gradient(variable):
...
+    Args:
+        x: A tensor. This function will try to access the `shape` attribute of
+            the input tensor.
+
+    Returns:
+        A tuple of integers or None values, indicating the shape of the input
+            tensor.
+
+    Examples:
+        >>> x = keras_core.zeros((8, 12))
```
Comment (fchollet) [inline keras_core/ops/core.py:287]: Unindent: code examples should be a the same level as the base docstring

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

**Comment by fchollet** [comment]
Also if there's a single example, use `Example:` as the section title

**Review by fchollet — APPROVED** [review]
Thanks!

**Pull request merged**